qiskit/ignis/experiments/calibration/methods/basic_1q.py

```diff
@@ -49,43 +58,56 @@ def __init__(self,
             freq_vals: Frequency values to scan in the calibration.
             analysis: Analysis class used.
             job: Optional job id to retrieve past experiments.
-            pulse_envelope: Name of the pulse function used to generate the
-                pulse schedule. If not specified, the default pulse shape of
-                :py:class:`SinglePulseGenerator` is used.
-            pulse_name: Pulse name in the database entry to provide parameter set to
-                construct pulse schedule to calibrate. By default pi pulse parameter is used.
+            spec_params: Parameters for the GaussianSquare spectroscopy pulse. If
+                None is given default values are assumed.
         """
+        if spec_params is None:
+            spec_params = {'duration': 3200, 'amp': 0.05, 'width': 0, 'sigma': 640}
+
+        delta = Parameter('freq')
+        with pulse.build(name='spec') as spec:
+            pulse.shift_frequency(delta, DriveChannel(qubit))
+            pulse.play(GaussianSquare(**spec_params), DriveChannel(qubit))
+
+        self.f0 = inst_def.backend.defaults().qubit_freq_est[qubit]
 
-        # todo get qubit property from other database.
-        # channel ref frequency is different from pulse sideband and thus
-        # this value is stored in another relational database.
-        # something like qubit property table where f01, anharmonicity, T1, T2, etc... exist.
-        freq01 = 0.0
+        qc = inst_def.get_circuit('spec', (qubit, ), schedule=spec)
+        qc.name = 'spec'
 
-        name = 'spectroscopy'
-        generator = CircuitBasedGenerator(
-            name=name,
-            qubits=[qubit],
-            template_circuit=inst_def.get_circuit(pulse_name, (qubit, )),
-            values_to_scan=freq_vals,
-            ref_frequency=freq01)
+        generator = CircuitBasedGenerator(self.name, [qubit], [qc], freq_vals)
+        analysis = self.default_analysis(self.name, data_processing, analysis, GaussianFit)
 
-        # setup analysis
-        if analysis is None:
-            analysis = GaussianFit(name=name)
+        super().__init__(self.name, inst_def, [], analysis, generator, job=job)
 
-        super().__init__(name, inst_def, [], analysis, generator, job)
+    def update_calibrations(self):
+        """
+        Updated frequency calibrations are stored in the backend as well as in the
+        pulse parameter table.
+        """
 
+        freq_offset = self.analysis.result[self.qubits[0]]['spec'].fitvals[1]
 
-class RoughAmplitudeCalibration(BaseCalibrationExperiment):
+        f_qubit = self.f0 + freq_offset
+
+        # Update the backend
+        self._inst_def.backend.defaults().qubit_freq_est[self.qubits[0]] = f_qubit
+
+        # Update the parameter table.
+        val = ParameterValue(f_qubit, datetime.now(), exp_id=self.job.job_id())
```

Comment:
I think we could but this would be a bit redundant with the job id. The current implementation lets the user know when the analysis was run and the parameter value updated.

Reply:
Note that analysis and execution are not necessary synchronized. We can submit the job and later extract experimental result with the execution id. Thus update time can be delayed for a long while.

Reply:
I think date time is more important than job id in terms of data analysis. I'm good with offloading this to future extension with TODO comment.

Reply:
Added TODO in 957acf7